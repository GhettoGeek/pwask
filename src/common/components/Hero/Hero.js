import React from 'react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import { makeStyles } from '@material-ui/core/styles'
import Typography from '../Typography'

const useStyles = makeStyles((theme) => ({
  root: {
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(2),
    backgroundSize: 'cover',
    backgroundRepeat: 'no',
  },
  title: {
    fontSize: '30px',
    fontWeight: 'bold',
    padding: theme.spacing(2),
  },
}))

function Hero({ backgroundImage, children }) {
  const classes = useStyles()
  const { t } = useTranslation()

  return (
    <div className={classes.root} style={{ backgroundImage: `url(${backgroundImage})` }}>
      <Typography className={classes.title} color="primary" align="center">
        {t('hero.title')}
      </Typography>
      {children}
    </div>
  )
}

Hero.propTypes = {
  backgroundImage: PropTypes.string,
  children: PropTypes.node,
}

Hero.defaultProps = {
  backgroundImage: '',
  children: '',
}

export default Hero
