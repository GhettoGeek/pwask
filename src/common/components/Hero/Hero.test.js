import React from 'react'
import { shallow } from 'enzyme'
import Hero from './Hero'

describe('Hero', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(<Hero />)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
