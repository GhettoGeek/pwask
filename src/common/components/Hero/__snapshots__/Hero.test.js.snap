// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly the component 1`] = `"<div class=\\"makeStyles-root-1\\" style=\\"background-image:url()\\"><h1 class=\\"MuiTypography-root makeStyles-title-2 MuiTypography-h1 MuiTypography-colorPrimary MuiTypography-alignCenter\\">hero.title</h1></div>"`;
